src/WebJobs.Script/Host/FunctionMetadataProvider.cs

```diff
@@ -170,6 +170,13 @@ internal static string ParseLanguage(string scriptFilePath, IEnumerable<RpcWorke
 
             // determine the script type based on the primary script file extension
             string extension = Path.GetExtension(scriptFilePath).ToLowerInvariant().TrimStart('.');
+            var workerConfig = workerConfigs.FirstOrDefault(config => config.Description.Extensions.Contains("." + extension));
```

Comment:
Unit tests? We have a few in FunctionMetadataProviderTests. Can you please add one?

Reply:
Good call. Added a test.